### PR TITLE
NIO-1180, requesting configuration version providing `api key` and `instance id`

### PR DESCRIPTION
**Pull request by f1401martin**
- notifying API with new ids once configuration is updated

### CODE REVIEW

File: manager.py [hunk 1]
```diff
@@ -66,31 +66,31 @@ def configure(self, context):
...
 
         # fetch instance specific settings
-        default = Persistence().load("api_key", default=None)
-        self.api_key = NIOEnvironment.get_variable('API_KEY', 
-                                                   default=default)
-
-        # config autonomy spcific settings
+        self._api_key = Persistence().load(
+            "api_key",
+            default=Settings.get("configuration", "instance_api_key"))
```
Comment (tlugger) [inline manager.py]: We would still want to have a way of pulling `API_KEY` from the NIOEnvironment for cloud instances

File: proxy.py [hunk 2]
```diff
@@ -11,61 +11,96 @@ def __init__(self):
...
+        """
+
+        url = "{}/instances/{}/configuration".format(url_prefix,
+                                                     instance_id)
+        return self._request(
+            requests.post, url, apikey, "Failed to post new configuration "
+                                        "version instance is running",
+            json={
+                "reported_configuration_id": config_id,
+                "reported_configuration_version_id": config_version_id
```
Comment (tlugger) [inline proxy.py]: Per API-444, this endpoint will also accept `deployment_id`, `status`, and `message` to update the product api when a deployment to an instance has succeeded or failed along with a message `updated [] services` or `failed to install [] block`

File: manager.py (hunk 1 above)
Comment (tlugger) [inline manager.py]: We would still want to have a way of pulling `API_KEY` from the NIOEnvironment for cloud instances

File: proxy.py [hunk 3]
```diff
@@ -11,61 +11,96 @@ def __init__(self):
...
+            url_prefix (str): url prefix to product api
+            instance_id (str): instance id
+            apikey (str): api key
 
         Returns: an object with format
             {
+                "instance_configuration_id": "uuid..",
                 "instance_configuration_version_id": "uuid..",
                 "version_num": "v1.0.0",
                 "created_at": "2018-03-14 12:12:12"
```
Comment (tlugger) [inline proxy.py]: `version_num` and `created_at` are no longer returned by this enpoint. Just the two UUIDs

File: proxy.py [hunk 4]
```diff
@@ -11,61 +11,96 @@ def __init__(self):
...
                 "configuration_data": {
                     "version": 1.0.0,
                     "blocks": {...},
                     "services": {...},
+                    "blockTypes": {...}
                 },
                 "message": "Found Instance Configuration...",
                 "status": 200,
                 "uuid": "uuid..",
                 "version_num": "v1.0.0"
```
Comment (tlugger) [inline proxy.py]: Additionally, `deployment_id` is returned at this endpoint

File: manager.py (hunk 1 above)
Comment (f1401martin) [reply manager.py]: The idea here is that the setting can read from the environment
```
[configuration]
instance_api_key=[[API_KEY]
...
[user_defined]
API_KEY = d400af7d-c725-47da-8ab7-67f7dc3fc067
```

Comment (tlugger) [reply manager.py]: Ahh, in that case this is good!

File: manager.py [hunk 5]
```diff
@@ -66,31 +75,31 @@ def configure(self, context):
...
-            fallback=None)
-        self._config_id = Persistence().load("configuration_id",
-                                            default=setting_config_id)
-        setting_config_version_id = \
-            Settings.get("configuration", "config_version_id",
-                fallback=None)
-        self._config_version_id = Persistence().\
-            load("configuration_version_id", default=setting_config_version_id)
-        
+                         fallback="https://api.n.io/v1/instance_configurations")
```
Comment (tlugger) [inline manager.py]: The fallback here should be chaned to `https://api.n.io/v1`

**Pull request merged**